src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -2334,9 +2334,13 @@ export class FileSystemError extends Error {
 		return new FileSystemError(messageOrUri, FileSystemProviderErrorCode.Unavailable, FileSystemError.Unavailable);
 	}
 
+	readonly code: string;
+
 	constructor(uriOrMessage?: string | URI, code: FileSystemProviderErrorCode = FileSystemProviderErrorCode.Unknown, terminator?: Function) {
 		super(URI.isUri(uriOrMessage) ? uriOrMessage.toString(true) : uriOrMessage);
 
+		this.code = code;
```

Comment:
I think it's better to use the name of the factory function itself, not rely on the enum that is spec outside of the API. Like so 👇 

```suggestion
		this.code = terminator?.name;
```

Reply:
It looks like the factory functions are almost never called in the VS Code source. Instead, the constructor is used, always without the `terminator` argument, but with a `code`.

https://github.com/microsoft/vscode/blob/1b70625f8c7c3b8ae24e22c25d261c19f99746e0/src/vs/workbench/api/common/extHostFileSystem.ts#L151

https://github.com/microsoft/vscode/blob/1b70625f8c7c3b8ae24e22c25d261c19f99746e0/src/vs/workbench/api/common/extHostFileSystem.ts#L151

Reply:
Yes, never! These are errors that are meant to be used in the API, e.g. from extensions, only. What might be missing is a proper conversion from  "internal" errors to these API error objects. That happens here: https://github.com/microsoft/vscode/blob/1b70625f8c7c3b8ae24e22c25d261c19f99746e0/src/vs/workbench/api/common/extHostFileSystem.ts#L139

Reply:
OK, got it. I've implemented the conversion by dispatching to appropriate factory functions in `_handleError()`.